rules/network/command_and_control_halfbaked_beacon.toml

```
@@ -0,0 +1,55 @@
+[metadata]
+creation_date = "2020/07/06"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/07/06"
+
+[rule]
+author = ["Elastic"]
+description = """
+Halfbaked is a malware family leveraged to establish persistence in a contested network. This rule detects a network
+activity algorithm leveraged by Halfbaked implant beacons for command and control.
+"""
+false_positives = [
+    """
+    This rule should be tailored to either exclude systems, as sources or destinations, in which this behavior is
+    expected.
+    """,
+]
+index = ["packetbeat-*"]
+language = "lucene"
+license = "Elastic License"
+name = "Halfbaked Command and Control Beacon"
+note = "This activity has been observed in FIN7 campaigns."
+references = [
+    "https://www.fireeye.com/blog/threat-research/2017/04/fin7-phishing-lnk.html",
+    "https://attack.mitre.org/software/S0151/",
+]
+risk_score = 73
+rule_id = "2e580225-2a58-48ef-938b-572933be06fe"
+severity = "high"
+tags = ["Elastic", "Network"]
+type = "query"
+
+query = '''
+event.category: network_traffic AND type:http AND network.transport: tcp AND url.full : /http:\/\/[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}\/cd/ AND destination.port:(53 OR 80 OR 8080 OR 443)
```

```suggestion
event.category:(network or network_traffic) AND type:http AND 
  network.transport:tcp AND url.full:/http:\/\/[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}\/cd/ AND 
  destination.port:(53 OR 80 OR 8080 OR 443)
```
adding `network` to `event.category` is for ECS compatibility - the stylistic changes are more of a nit, so feel free to disregard

---

Question @peasead: is the intent to always see an IP address or is it more about the `cd` path? The reason I ask is because you can also split it up to test `url.path:"/cd"` but if the IP is needed, then you can do `url.domain:<ip-regex>`  if needed. This would also account for https. 

Thoughts? There are a couple ways to do this, so just lmk

---

Yes, I did think about splitting it up for TLS, you don't see the URI (`/cd`) over https, so I decided not to as `https://x.x.x.x` would be super noisy.

It is `http://x.x.x.x:{53,80,8080,443}/cd` (IP + `cd`), but `packetbeat-*` doesn't grab `:port`, so I had to use `http://x.x.x.x/cd` and use `destination.port`.